validation/pytorch.py

```diff
@@ -40,16 +41,21 @@
 
 def self_consistency(rnn, x):
   x_cuda = x.clone().cuda()
-  x_cpu = x.clone().cpu()
+  x_cuda_torch = x_cuda.detach().clone()
   x_cuda.requires_grad_(True)
-  x_cpu.requires_grad_(True)
+  x_cuda_torch.requires_grad_(True)
 
-  y1, _ = rnn.cuda().forward(x_cuda)
+  rnn.cuda()
+  seed = 5566
+  torch.manual_seed(seed)
```

Comment:
It's probably a good idea to control the seed for reproducible test results. Maybe not hard-coded in here like this, though? I suggest either removing the RNG seeding, or setting the RNG seed through a command line argument with a default of no-explicit-seed.

Reply:
We set the seed manually to make sure that the two runs will use identical random masks (*zoneout* masks and *dropout* masks) generated inside `forward`. Maybe the seed could be some random value hard-coded like [pytorch tests](https://github.com/pytorch/pytorch/blob/20489ebdc92068823941de285474f9b5a745c68d/test/test_cuda.py#L656-L664)?

Reply:
Ah, this makes sense – you're re-seeding between the two runs to generate the same random masks, and not to just get reproducible results across multiple runs of this script.